pandas/tests/series/methods/test_truncate.py

```diff
@@ -80,6 +80,18 @@ def test_truncate_nonsortedindex(self):
         with pytest.raises(ValueError, match=msg):
             ts.sort_values(ascending=False).truncate(before="2011-11", after="2011-12")
 
+    @pytest.mark.parametrize(
+        "before, after, indices",
+        [(1, 2, [2, 1]), (None, 2, [2, 1, 0]), (1, None, [3, 2, 1])],
+    )
+    def test_truncate_decreasing_index(self, before, after, indices):
+        # https://github.com/pandas-dev/pandas/issues/33756
+        idx = pd.Index([3, 2, 1, 0])
+        values = pd.Series(index=idx)
+        result = values.truncate(before=before, after=after)
+        expected = values.loc[indices]
+        tm.assert_series_equal(result, expected)
```

Comment:
This code duplication isn't pretty, should the truncate methods instead be tested in a more general place and parametrized?

Reply:
for methods where we can share+parametrize the tests over Series/DataFrame, we've been putting those in tests.generic.methods

Reply:
should have been more clear: so far at least we are only doing this for methods where we can do this for _all_ of the tests, so that we dont end up with 3 separate test files.

If we're not yet ready to combine all the tests, lets live with the duplication for now.

most of the methods.test_foo files need eyeballs for some combination of a) de-duplication and b) split up giant legacy tests